server/server_app.js

```diff
@@ -95,12 +98,19 @@ app.get('/about.json', async (req, res) => {
  * body.username -> User name
  * body.email -> User mail
  * body.password -> User password
+ * An e-mail is now send to the user.
  */
 app.post('/api/signup', (req, res, next) => {
   passport.authenticate('signup', { session: false }, (err, user, info) => {
     if (err) throw new Error(err)
     if (user == false) return res.json(info)
     const token = utils.generateToken(user.id)
+    gmail.sendEmail(
```

Comment:
missing check if email has been send ; 

```
gmail_service.sendEmail(to, subject, body)
  .then((response) => {
    console.log('Email sent successfully:', response);
  })
  .catch((error) => {
    console.error('Error sending email:', error);
  });
```

replace the console.log by handling the error

Reply:
although the email have already been verificated looking through the previous code, just catch error in case